easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -472,47 +472,57 @@ def dump(self, fp):
         """
         eb_file = file(fp, "w")
 
-        def to_str(x):
-            """Return quoted version of x"""
-            if isinstance(x, basestring):
-                if '\n' in x or ('"' in x and "'" in x):
-                    return '"""%s"""' % x
-                elif "'" in x:
-                    return '"%s"' % x
-                else:
-                    return "'%s'" % x
-            else:
-                return "%s" % x
-
         # ordered groups of keys to obtain a nice looking easyconfig file
         grouped_keys = [
+            ['easyblock'],
             ['name', 'version', 'versionprefix', 'versionsuffix'],
             ['homepage', 'description'],
             ['toolchain', 'toolchainopts'],
-            ['source_urls', 'sources'],
+            ['sources', 'source_urls'],
             ['patches'],
             ['builddependencies', 'dependencies', 'hiddendependencies'],
+            ['osdependencies'],
+            ['preconfigopts', 'configopts'],
+            ['prebuildopts', 'buildopts'],
+            ['preinstallopts', 'installopts'],
             ['parallel', 'maxparallel'],
-            ['osdependencies']
         ]
 
+        last_keys = [
+            ['sanity_check_paths'],
+            ['moduleclass'],
+        ]
+
+        def check_and_print(keyset):
+            """
+            Internal function checking for default values
+            """
+            for group in keyset:
+                printed = False
+                for key1 in group:
+                    val = self._config[key1][0]
+                    for key2, [def_val, _, _] in DEFAULT_CONFIG.items():
+                        # only print parameters that are different from the default value
+                        if key1 == key2 and val != def_val:
+                            ebtxt.append("%s = %s" % (key1, quote_str(val, escape_newline=True)))
+                            printed_keys.append(key1)
+                            printed = True
+                if printed:
+                    ebtxt.append("")
+
+
         # print easyconfig parameters ordered and in groups specified above
         ebtxt = []
         printed_keys = []
-        for group in grouped_keys:
-            for key1 in group:
-                val = self._config[key1][0]
-                for key2, [def_val, _, _] in DEFAULT_CONFIG.items():
-                    # only print parameters that are different from the default value
-                    if key1 == key2 and val != def_val:
-                        ebtxt.append("%s = %s" % (key1, to_str(val)))
-                        printed_keys.append(key1)
-            ebtxt.append("")
+        check_and_print(grouped_keys)
 
         # print other easyconfig parameters at the end
         for key, [val, _, _] in DEFAULT_CONFIG.items():
```

Comment:
fix the order in which the keys are being considered here (the order of the elements in `.items()` on a dict-like object may differ, you can't make assumptions on it's order here); maybe used `sorted`?